bundles/org.openhab.binding.zm/src/main/java/org/openhab/binding/zm/internal/handler/ZmBridgeHandler.java

```diff
@@ -0,0 +1,648 @@
+        try (ByteArrayInputStream inputStream = new ByteArrayInputStream(content.getBytes())) {
+            long startTime = System.currentTimeMillis();
+            String response = HttpUtil.executeUrl("POST", url, inputStream, contentType, API_TIMEOUT_MSEC);
+            logger.trace("Bridge: Http POST content '{}' to '{}' returned: {} in {} ms", content, url, response,
+                    System.currentTimeMillis() - startTime);
+            return response;
+        } catch (IOException e) {
+            logger.debug("Bridge: IOException: {}", e.getMessage(), e);
+        }
+        return null;
+    }
+
+    public String buildLoginUrl() {
+        return buildBaseUrl(LOGIN_PATH).toString();
+    }
+
+    public String buildLoginUrl(String tokenParameter) {
+        StringBuilder sb = buildBaseUrl(LOGIN_PATH);
+        sb.append(tokenParameter);
+        return sb.toString();
+    }
+
+    private String buildStreamUrl(String id, String streamType) {
+        List<String> parameters = new ArrayList<>();
+        parameters.add(String.format("mode=%s", streamType));
+        parameters.add(String.format("monitor=%s", id));
+        return buildUrlWithParameters("/cgi-bin/zms", parameters);
+    }
+
+    private String buildUrl(String path) {
+        return buildUrlWithParameters(path, EMPTY_LIST);
+    }
+
+    private String buildUrlWithParameters(String path, List<String> parameters) {
+        StringBuilder sb = buildBaseUrl(path);
+        String joiner = "?";
+        for (String parameter : parameters) {
+            sb.append(joiner).append(parameter);
+            joiner = "&";
+        }
+        if (zmAuth.usingAuthorization()) {
+            sb.append(joiner).append("token=").append(zmAuth.getAccessToken());
+        }
+        return sb.toString();
+    }
+
+    private StringBuilder buildBaseUrl(String path) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("http://");
+        sb.append(host);
+        if (portNumber != null) {
+            sb.append(":").append(portNumber);
+        }
+        if (useDefaultUrlPath) {
+            sb.append("/zm");
+        } else if (urlPath != null) {
+            sb.append(urlPath);
+        }
+        sb.append(path);
+        return sb;
+    }
+
+    private boolean isMonitorIdValid(String id) {
+        return savedMonitors.stream().filter(monitor -> id.equals(monitor.getId())).findAny().isPresent();
+    }
+
+    private boolean isHostValid() {
+        logger.debug("Bridge: Checking for valid Zoneminder host: {}", host);
+        VersionDTO version = getVersion();
+        if (version != null) {
+            if (checkSoftwareVersion(version.version) && checkApiVersion(version.apiVersion)) {
+                return true;
+            }
+        } else {
+            logger.warn("Bridge: Can't get version information from server");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Can't get version information");
```

Comment:
Can the log message be removed? See point 5 https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
I found this to be useful when debugging user issues, so I would prefer to keep. I downgraded it to INFO level.

Reply:
The custom configuration error message, supplied to `updateStatus()` is logged by the framework. The user simply has to increase the log level. I think there is no need for logging the same message to different loggers.

Reply:
Ok